react/tic-tac-toe/src/redux/login/actions.js

```diff
@@ -1,15 +1,27 @@
 import LoginService  from 'services/LoginService';
 import api from 'config/api'
+import { createTypes, completeTypes } from 'redux-recompose';
 import actionsGeneral from '../general/actions';
 
-export const actions = {
-  GET_TOKEN: '@@LOGIN/GET_TOKEN',
-  GET_TOKEN_SUCCESS: '@@LOGIN/GET_TOKEN_SUCCESS',
-  GET_TOKEN_FAILURE: '@@LOGIN/GET_TOKEN_FAILURE'
-}
+export const actions = createTypes(completeTypes(['GET_TOKEN']), '@@LOGIN');
+
+const privateActionCreators = {
+  getTokenSuccess: values => ({
+    type: actions.GET_TOKEN_SUCCESS,
+    payload: values
+  }),
+  getTokenError: error => ({
+    type: actions.GET_TOKEN_SUCCESS,
+    payload: error
+  })
+};
 
 const actionsCreators = {
   getToken: (body) => async dispatch => {
+    dispatch({ 
+      type: actions.GET_TOKEN,
+      target: 'token'
+     });
     const response = await LoginService.getToken(body);
```

Comment:
Check [this](https://github.com/Wolox/redux-recompose/tree/master/src/middlewares) feature out! It can simplify the logic you're using here. :)